App.java

```diff
@@ -0,0 +1,122 @@
+    private static final List<String> SCOPES =
+        Arrays.asList(SheetsScopes.SPREADSHEETS_READONLY);
+
+    static {
+        try {
+            HTTP_TRANSPORT = GoogleNetHttpTransport.newTrustedTransport();
+            DATA_STORE_FACTORY = new FileDataStoreFactory(DATA_STORE_DIR);
+        } catch (Throwable t) {
+            t.printStackTrace();
+            System.exit(1);
+        }
+    }
+
+    /**
+     * Creates an authorized Credential object.
+     * @return an authorized Credential object.
+     * @throws Exception 
+     */
+    public static Credential authorize() throws Exception {
+        // Load client secrets.
+        InputStream in =
+           App.class.getResourceAsStream("/client_secert.json");
+        GoogleClientSecrets clientSecrets =
+            GoogleClientSecrets.load(JSON_FACTORY, new InputStreamReader(in));
+
+        // Build flow and trigger user authorization request.
+        GoogleAuthorizationCodeFlow flow =
+                new GoogleAuthorizationCodeFlow.Builder(
+                        HTTP_TRANSPORT, JSON_FACTORY, clientSecrets, SCOPES)
+                .setDataStoreFactory(DATA_STORE_FACTORY)
+                .setAccessType("offline")
+                .build();
+        Credential credential = new AuthorizationCodeInstalledApp(
+            flow, new LocalServerReceiver()).authorize("user");
+        System.out.println(
+                "Credentials saved to " + DATA_STORE_DIR.getAbsolutePath());
+        return credential;
+    }
+
+    /**
+     * Build and return an authorized Sheets API client service.
+     * @return an authorized Sheets API client service
+     * @throws Exception 
+     */
+    public static Sheets getSheetsService() throws Exception {
+        Credential credential = authorize();
+        return new Sheets.Builder(HTTP_TRANSPORT, JSON_FACTORY, credential)
+                .setApplicationName(APPLICATION_NAME)
+                .build();
+    }
+
+    public static void main(String[] args) throws Exception {
+        // Build a new authorized API client service.
+        Sheets service = getSheetsService();
+
+        // Prints the names and majors of students in a sample spreadsheet:
+        // https://docs.google.com/spreadsheets/d/1BxiMVs0XRA5nFMdKvBdBZjgmUUqptlbs74OgvE2upms/edit
+        String spreadsheetId = "1BxiMVs0XRA5nFMdKvBdBZjgmUUqptlbs74OgvE2upms";
+        String range = "Class Data!A2:E";
+        ValueRange response = service.spreadsheets().values()
+            .get(spreadsheetId, range)
+            .execute();
+        List<List<Object>> values = response.getValues();
+        if (values == null || values.size() == 0) {
+            System.out.println("No data found.");
+        } else {
+          System.out.println("Name, Major");
+          for (List row : values) {
+            // Print columns A and E, which correspond to indices 0 and 4.
+            System.out.printf("%s, %s\n", row.get(0), row.get(4));
+          }
+        }
+    }
+
+
+}
```

Comment:
//access USD edition name , offer id, price

 public static void main(String[] args) throws Exception {
        // Build a new authorized API client service.
        Sheets service = getSheetsService();

        // Prints the names and majors of students in a sample spreadsheet:
        // https://docs.google.com/spreadsheets/d/1BxiMVs0XRA5nFMdKvBdBZjgmUUqptlbs74OgvE2upms/edit
        String spreadsheetId = "11TeFVUDVKt1AhQJRIgICb2Hx-EJfVhUNHCtPisagGOc";
        String range = "USD!3:100";
        ValueRange response = service.spreadsheets().values()
            .get(spreadsheetId, range)
            .execute();
        List<List<Object>> values = response.getValues();
        if (values == null || values.size() == 0) {
            System.out.println("No data found.");
        } else {
           System.out.println("Edition Name,  Edition id ,Price");

         for (List row : values) {
              System.out.printf("%s, %s ,%s \n", row.get(3), row.get(4), row.get(10));
            }
            
          
        }
    }
}